File: app/src/main/java/net/n4dev/treespot/db/UUIDConverter.java
package net.n4dev.treespot.db;

import java.util.UUID;

import io.objectbox.converter.PropertyConverter;

public class UUIDConverter implements PropertyConverter<UUID, String> {

    @Override
    public UUID convertToEntityProperty(String databaseValue) {
        return UUID.fromString(databaseValue);
    }

    @Override
    public String convertToDatabaseValue(UUID entityProperty) {
        return entityProperty.toString();
    }
}

File: app/src/main/java/net/n4dev/treespot/db/UUIDConverter.kt
package net.n4dev.treespot.db

import io.objectbox.converter.PropertyConverter
import java.util.*

class UUIDConverter : PropertyConverter<UUID, String?> {
    override fun convertToEntityProperty(databaseValue: String?): UUID {
        return UUID.fromString(databaseValue)
    }

    override fun convertToDatabaseValue(entityProperty: UUID): String {
        return entityProperty.toString()
    }
}